packages/block-library/src/image/edit.js

```diff
@@ -275,7 +275,7 @@ export function ImageEdit( {
 	return (
 		<>
 			{ controls }
-			<Block.figure ref={ ref } className={ classes } key={ key }>
+			<figure { ...useBlockProps( { key, ref, className: classes } ) }>
```

Comment:
I think it's fine to leave `key` as it was, it's used only to help React with re-renders.

Reply:
I've removed it now. We should avoid triggering a remount.